packages/storybook/stories/CustomIcons/utils.tsx

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+import { mergeStyles, Stack, TextField } from '@fluentui/react';
+import React, { useState } from 'react';
+
+const iconGrid = (allIcons: Record<string, JSX.Element>, searchText?: string): JSX.Element[] => {
+  const allIconKeys = Object.keys(allIcons);
+  const filteredKeys = searchText
+    ? allIconKeys.filter((k) => k.toLowerCase().includes(searchText.toLowerCase()))
+    : allIconKeys;
+
+  return filteredKeys.map((key, idx) => {
+    return (
+      <Stack
+        key={idx}
+        horizontal
+        style={{
+          padding: '0.5rem 0'
+        }}
+      >
+        <Stack horizontalAlign="start" className={mergeStyles({ span: { fontSize: '1rem' } })}>
+          {allIcons[key]}
+        </Stack>
+        <Stack
+          horizontalAlign="start"
+          style={{
+            paddingLeft: '0.5rem',
+            fontSize: '0.75rem',
+            fontFamily: 'consolas'
+          }}
+        >
+          {key}
+        </Stack>
+      </Stack>
+    );
+  });
+};
+
+const SearchBar = (props: { onChange: (newValue?: string) => void }): JSX.Element => {
+  const { onChange } = props;
+  return (
+    <Stack style={{ padding: '1rem 0' }} horizontalAlign="start">
+      <TextField
+        label="Search Icons:"
+        underlined
+        onChange={(e, val) => {
+          onChange(val);
+        }}
+      />
+    </Stack>
+  );
+};
+
+export const IconGridWithSearch = (props: { icons: Record<string, JSX.Element> }): JSX.Element => {
+  const { icons } = props;
+  const [searchText, setSearchText] = useState<string | undefined>(undefined);
+  return (
+    <Stack>
+      <SearchBar onChange={(val) => setSearchText(val)} />
+      <div style={{ display: 'grid', gridTemplateColumns: 'auto auto auto', padding: '1rem 0' }}>
```

Comment:
I keep going to read the icons top-bottom then left-right. Can you confirm which is the preferred UX?
I like the argument here that because there is more space between columns then rows, and because we read grouped things together first, the columns appear to us as a grouping and thus we read top->bottom: https://stackoverflow.com/a/3250389/16089184

However, jury is definitely out on this one - so don't mind going with either

If you did want to sort top->bottom then left->right the fluent stack can do this; see "Vertical Stack - Wrapping - Nested" in https://developer.microsoft.com/en-us/fluentui#/controls/web/stack

Reply:
Here is possibly the most famous icon library out there. Took inspiration from it.
https://fontawesome.com/v4.7/icons/
It is following Left -> Right and then Top -> Bottom
I personally read them left to right as intended. Pretty sure once a user looks at them, they will identify the pattern and wouldn't find it hard to navigate.

That's another reason I added search